src/common/SearchInput.js

```diff
@@ -14,32 +16,52 @@ const styles = StyleSheet.create({
     borderWidth: 0,
     // color: BRAND_COLOR,
   },
+  icon: {
+    margin: 6,
+  }
 });
 
+type Props = {
+  onChange: (text: string) => void,
+  showCancelIcon: boolean,
+};
+
 export default class SearchInput extends Component {
 
-  props: {
-    onChange: (text: string) => void,
-  };
+  props: Props;
+  textInput: TextInput;
+
+  handleClear = () => {
+    this.textInput.clear();
+    this.props.onChange('');
+  }
 
   render() {
-    const { onChange } = this.props;
+    const { onChange, showCancelIcon } = this.props;
```

Comment:
Move the `showCancelIcon` to <Input /> ?
Also call it `clearButton` as this matches the iOS version naming.